sdv/multi_table/base.py

```diff
@@ -368,6 +368,35 @@ def get_learned_distributions(self, table_name):
         synthesizer = self._table_synthesizers[table_name]
         return synthesizer.get_learned_distributions()
 
+    def _validate_constraints(self, constraints):
+        for constraint_dict in constraints:
+            params = {'constraint_class', 'constraint_parameters', 'table_name'}
+            keys = constraint_dict.keys()
+            missing_params = params - keys
+            if missing_params:
+                raise SynthesizerInputError(
+                    f'A constraint is missing required parameters {missing_params}. '
+                    'Please add these parameters to your constraint definition.'
+                )
+
+            extra_params = keys - params
+            if extra_params:
+                raise SynthesizerInputError(
+                    f'Unrecognized constraint parameter {extra_params}. '
+                    'Please remove these parameters from your constraint definition.'
+                )
```

Comment:
It feels like we are doing this twice, I think we can just check if `table_name` is not in `constraint_dict` raise the error stating that you are missing `table_name`, if not then the data processor will raise the error.

Reply:
Yeah, I did it like that so the errors would show up together (if you don't pass any parameters, it will crash saying only `table_name` is missing, and only after you add `table_name` will it tell you the other errors).

But I think your way is better, it's too much duplicated code otherwise.